text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,1124 @@
+2:03:21 PM | DELETE_COMPLETE      | Custom::DynamoDBReplica             | MyStack/MyTableReplicauswest285A33668/Resource
+2:03:22 PM | DELETE_COMPLETE      | AWS::IAM::ManagedPolicy             | MyStack/MyTableManagedPolicyXXXXXXX/Resource
+2:03:22 PM | DELETE_COMPLETE      | AWS::CloudFormation::NestedStack    | MyStack/ReplicaProviderNestedStack/Resource
+2:03:19 PM | DELETE_COMPLETE      | AWS::DynamoDB::Table                | MyStack/MyTable794EDED1/Resource
+
+✅  Stack deployment complete
+```
+
+> For CI/CD use cases, you can streamline deployments by using the `--require-approval never`
+> option with the CDK CLI to bypass manual approval prompts.
+
+### Drift Detection Failure Example
+
+If a CloudFormation drift is detected, the CLI will show:
+
+```s
+Performing construct upgrade validation...
+
+⚠️  Note: This is an experimental feature
+
+Validations:
+❌ CloudFormation drift detected
+  │ Resource: AWS::DynamoDB::Table MyTable794EDED1
+  │ Actual changes found:
+  └   • BillingMode: "PAY_PER_REQUEST" (expected: "PROVISIONED")
+✅ Change set analysis complete
+✅ Retain policies configured
+✅ Construct-specific validations passed
+
+Error: Construct upgrade validation failed
+Please resolve the drift before proceeding with the upgrade.
+For more information about handling drift, visit: 'https://docs.aws.amazon.com/cdk/v2/guide/drift'
+```
+
+### Change Set Validation Failure Example
+
+If the change set analysis detects the new table is not being imported, the CLI will show:
+
+```s
+Performing construct upgrade validation...
+
+⚠️  Note: This is an experimental feature
+
+Validations:
+✅ No CloudFormation drift detected
+❌ Change set analysis failed
+  │ Resources: MyGlobalTable794EDED1 (AWS::DynamoDB::GlobalTable)
+  │ Change Set:
+  └  • Action: 'Add' (expected: 'Import')
+✅ Retain policies configured
+✅ Construct-specific validations passed
+
+Error: Construct upgrade validation failed
+```
+
+### Retain Policies Failure Example
+
+Here's the error output example for when retain policies validation fails:
+
+```s
+Performing construct upgrade validation...
+
+⚠️  Note: This is an experimental feature
+
+Validations:
+✅ No CloudFormation drift detected
+✅ Change set analysis complete
+❌ Retain policies validation failed
+  │ Resource: MyTable794EDED1 (AWS::DynamoDB::Table)
+  └   • RemovalPolicy: 'Delete' (expected: 'Retain')
+✅ Construct-specific validations passed
+
+Error: Construct upgrade validation failed
+```
+
+#### Construct-specific Validation Failure Example
+
+If construct-specific validation fails, the CLI will show:
+
+```s
+Performing construct upgrade validation...
+
+⚠️  Note: This is an experimental feature
+
+Validations:
+✅ No CloudFormation drift detected
+✅ Change set analysis complete
+✅ Retain policies configured
+❌ Construct-specific validations failed
+  │ Resources: MyGlobalTable794EDED1 (AWS::DynamoDB::GlobalTable)  │ 
+  │ Property 'TableName' is required for import but was not specified
```

Comment:
If `TableName` is not configured, then change set analysis will fail - do we really need this validation then? will we even get here? It seems more like a cause rather than its own validation. 

Also please add the `SkipReplicationDeletion` validation.